### DISCUSSION TITLE
Is it possible to deploy code-server on to a cloud service?

### DISCUSSION THREAD

**Discussion by AyaanZaveri** [coder/code-server#3628] [General] [answered]
Is it possible to deploy it on to a cloud service like Heroku or Firebase? Also, is it possible to deploy it on a CI like CircleCI. I'd really like if that would be possible, it's easy to access since it's on the cloud. Thank you!

**Comment by jsjoeio** [comment, marked as answer]
Yes, it's possible. Please see [this repo](https://github.com/cdr/deploy-code-server) for all the options for deploying code-server. 

> deploy it on a CI like CircleCI

Based on how I'm understanding your question, I don't think this is possible. Though feel free to elaborate and I'll see if I can help!

**Reply by AyaanZaveri** [reply]
Thanks! On CircleCI is it possible to run [this](https://hub.docker.com/r/linuxserver/code-server) docker container, and then use [localtunnel](https://github.com/localtunnel/localtunnel) to get a url?

**Reply by jsjoeio** [reply]
Hmm...I think it should be if you're using it for testing/CI but not for deploying and using code-server, but I haven't used CircleCI in a long time.

localtunnel should work to get a url too. I've used [ngrok](https://ngrok.com/) in the past, but sounds similar. 

cc @bpmct might have better suggestions too!

**Reply by AyaanZaveri** [reply]
Thanks for your help!
